string outMessage
int16 size
---
string inMessage
